plugins/org.jkiss.dbeaver.net.ssh/src/org/jkiss/dbeaver/model/net/ssh/SSHUtils.java

```diff
@@ -78,4 +90,94 @@ public static boolean isKeyEncrypted(byte[] privKeyValue) {
         return false;
     }
 
+    private static final String PLATFORM_SSH_PREFERENCES_NODE = "org.eclipse.jsch.core"; //$NON-NLS-1$
+    private static final String PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY = IConstants.KEY_SSH2HOME;
+    private static final String DEFAULT_SSH_HOME_DIR_NAME = IConstants.SSH_DEFAULT_HOME;
+    private static final String DEFAULT_SSH_HOME_DIR_NAME_WIN_OLD = IConstants.SSH_OLD_DEFAULT_WIN32_HOME;
+    private static final String KNOWN_SSH_HOSTS_FILE_NAME = "known_hosts";
+
+    @NotNull
+    public static File getKnownSshHostsFileOrDefault() {
+        return  getKnownSshHostsFileImpl(true);
+    }
+
+    @Nullable
+    public static File getKnownSshHostsFileOrNull() {
+        return  getKnownSshHostsFileImpl(false);
+    }
+
+    private static File getKnownSshHostsFileImpl(boolean forceFileObjectOnFail) {
```

Comment:
Please annotate even non-public API methods with `@NotNull`/`@Nullable` annotations.

Reply:
The nullability of that two methods depends on argument. Which of these annotations should I use in this case?

Reply:
Maybe the caller should handle exceptional situations themselves?

Let's not pass the `forceFileObjectOnFail` parameter and let the caller decide how to handle this situation.

So it would become `@Nullable`.